src/Hng.Web/Controllers/AdminController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Hng.Application.Features.SuperAdmin.Dto;
+using Hng.Application.Features.SuperAdmin.Queries;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [ApiController]
+    [Route("api/v1/admin")]
+    public class AdminController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        public AdminController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        /// <summary>
+        /// Admin: Users - gets all registered users by search parameters
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("users")]
+        [Authorize]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> GetUsersBySearch([FromQuery] UsersQueryParameters parameters)
+        {
+            var users = await _mediator.Send(new GetUsersBySearchQuery(parameters));
+            return Ok(new
+            {
+                status_code = 200,
+                users.MetaData,
+                users
+            });
```

Comment:
Please check the HNG application, there's a folder called shared, it has our generic response dto, return that and set the data field to this data you want to return

Reply:
The generic response dto does not satisfy the response data structure. this is what the response looks like when using the generic response dto

{
  "data": [
    {
      "id": "da037533-179f-434c-bb24-6ab966315672",
      "first_name": "Burdette",
      "last_name": "Grimes",
      "avatar_url": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/33.jpg",
      "email": "Burdette0@hotmail.com",
      "created_on": "0001-01-01T00:00:00"
    },
    {
      "id": "f2755464-6326-47ad-9cff-7d5e62035599",
      "first_name": "Lemuel",
      "last_name": "Schneider",
      "avatar_url": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1151.jpg",
      "email": "Lemuel.Schneider33@yahoo.com",
      "created_on": "0001-01-01T00:00:00"
    }
  ],
  "message": "Request completed successfully."
}

whereas, the response should look like this 

{
  "status_code": 200,
  "metaData": {
    "totalPages": 1,
    "currentPage": 1,
    "pageSize": 10,
    "totalCount": 2,
    "hasPrevious": false,
    "hasNext": false
  },
  "users": [
    {
      "id": "da037533-179f-434c-bb24-6ab966315672",
      "first_name": "Burdette",
      "last_name": "Grimes",
      "avatar_url": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/33.jpg",
      "email": "Burdette0@hotmail.com",
      "created_on": "0001-01-01T00:00:00"
    },
    {
      "id": "f2755464-6326-47ad-9cff-7d5e62035599",
      "first_name": "Lemuel",
      "last_name": "Schneider",
      "avatar_url": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1151.jpg",
      "email": "Lemuel.Schneider33@yahoo.com",
      "created_on": "0001-01-01T00:00:00"
    }
  ]
}

Reply:
Okay let's have this.
In the data for the generic response pass new{metadata, user}

Reply:
Ok, what i'll do is create a generic response for paginated data

Reply:
Sweet!